aspnetcore/fundamentals/middleware/index.md

```diff
@@ -10,7 +10,284 @@ uid: fundamentals/middleware/index
+| [DeveloperExceptionPage](xref:Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware) | Generates a page with error information that is intended for use only in the Development environment. | Before components that generate errors. The project templates automatically register this middleware as the first middleware in the pipeline when the environment is Development. |
+| [Diagnostics](xref:fundamentals/error-handling) | Several separate middlewares that provide a developer exception page, exception handling, status code pages, and the default web page for new apps. | Before components that generate errors. Terminal for exceptions or serving the default web page for new apps. |
+| [Forwarded Headers](xref:host-and-deploy/proxy-load-balancer) | Forwards proxied headers onto the current request. | Before components that consume the updated fields. Examples: scheme, host, client IP, method. |
+| [Health Check](xref:host-and-deploy/health-checks) | Checks the health of an ASP.NET Core app and its dependencies, such as checking database availability. | Terminal if a request matches a health check endpoint. |
+| [Header Propagation](xref:fundamentals/http-requests#header-propagation-middleware) | Propagates HTTP headers from the incoming request to the outgoing HTTP Client requests. |
+| [HTTP Logging](xref:fundamentals/http-logging/index) | Logs HTTP Requests and Responses. | At the beginning of the middleware pipeline. |
+| [HTTP Method Override](xref:Microsoft.AspNetCore.Builder.HttpMethodOverrideExtensions) | Allows an incoming POST request to override the method. | Before components that consume the updated method. |
+| [HTTPS Redirection](xref:security/enforcing-ssl#require-https) | Redirect all HTTP requests to HTTPS. | Before components that consume the URL. |
+| [HTTP Strict Transport Security (HSTS)](xref:security/enforcing-ssl#http-strict-transport-security-protocol-hsts) | Security enhancement middleware that adds a special response header. | Before responses are sent and after components that modify requests. Examples: Forwarded Headers, URL Rewriting. |
+| [MVC](xref:mvc/overview) | Processes requests with MVC/Razor Pages. | Terminal if a request matches a route. |
+| [OWIN](xref:fundamentals/owin) | Interop with OWIN-based apps, servers, and middleware. | Terminal if the OWIN Middleware fully processes the request. |
+| [Output Caching](xref:performance/caching/output) | Provides support for caching responses based on configuration. | Before components that require caching. `UseRouting` must come before `UseOutputCaching`.|
+| [Request Decompression](xref:fundamentals/middleware/request-decompression) | Provides support for decompressing requests. | Before components that read the request body. |
+| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. | Before components that require caching. `UseCORS` must come before `UseResponseCaching`.|
```

Comment:
Can we add some text here to discourage using response caching unless that's really what you want? It also probably makes sense to put "Output Caching" next to "Response Caching" if we're not doing this alphabetically anymore.

```suggestion
| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. This requires client participation to work. Use output caching for complete server control. | Before components that require caching. `UseCORS` must come before `UseResponseCaching`.|
| [Request Decompression](xref:fundamentals/middleware/request-decompression) | Provides support for decompressing requests. | Before components that read the request body. |
```

Reply:
> if we're not doing this alphabetically anymore

Side idea that could result in opening a follow-up issue to this PR for work later ...

It seems fine to have a list in this topic that's restructured to reflect related middleware, but also consider that this list is ***buried*** ⛏️ to a great degree, which makes it somewhat difficult to get to a specific middleware's coverage quickly.

For Razor components, I addressed this problem by not listing them out in the **Blazor** > **Components** > **Overview** topic. I created a ToC entry at the end of the **Blazor** > **Components** ToC node for a topic ...

![image](https://user-images.githubusercontent.com/1622880/203307645-ce2f11f6-49ab-4d78-8819-b00c8766fd5c.png)

... which is at ... 

https://learn.microsoft.com/en-us/aspnet/core/blazor/components/built-in-components?view=aspnetcore-7.0

... and that's where I keep an alphabetical list for easy navigation when trying to track one down ... no crazy scrolling or opening the **hidden** 😠 *Overview* topic ToC (btw ... that's in the ***black hole*** ⚫ at [GH issue](https://github.com/MicrosoftDocs/feedback/issues/3770) in spite of 25 upvotes) ... anyway ... no problems for readers trying to find the list/links in the *Overview* topic.

In case you're wondering why I didn't just place them into the ToC, there are two reasons:

* The ToC highlighting and back-button navigation goes by the landed page, so it can lead to poor highlighting and navigation, particularly when trying to read our docs in order top-to-bottom. This is also true for tutorials, which is why I use [a ***topic*** to list the tutorials](https://learn.microsoft.com/en-us/aspnet/core/blazor/tutorials/?view=aspnetcore-7.0) and link to the topic from the ToC.
* The ToC is too ***stable***. By using a topic, I can version the list easily release to release.

The same approach could be taken for middleware.